packages/babel-preset-fbjs/plugins/inline-requires.js

```diff
@@ -68,8 +77,14 @@ module.exports = babel => ({
               if (parseResult == null) {
                 return;
               }
-              const {declarationPath, moduleName} = parseResult;
 
+              // Do not transform require calls if require is redeclared in global scope
+              const requireBinding = path.scope.getBinding('require');
+              if (requireBinding != null && requireBinding.scope.parent == null) {
+                return;
+              }
+
+              const {declarationPath, moduleName} = parseResult;
```

Comment:
I don't think we should limit this check to the `global` scope only. E.g. we shouldn't perform any modification whenever a user calls a custom `require` function. E.g. doing any changes wouldn't be desired in the following example (worth adding a test case if it's missing). 

```
function test () {
  function require(condition) {
    if (!condition) {
      throw new Error(`The condition is false`);
    }
  }

  require('test');
}
```

There's another complexity that on top of this. The plugin allows consumers to pass custom `require` function names in `state.inlineableCalls`. Metro depends on this logic because it first lowers `import * from 'x'` and `import X from 'x'x` to `const x = importAll('x')` and `const X = importDefault('x')` calls that should then be inlined by this inline plugin (yes, this is not 100% spec conform). Metro will use unique names for the `importAll` and `importDefault` functions so it's technically not a problem but we shouldn't make assumptions in this plugin on how it's used.

That means that you can't test for a `require` bindings only but instead must test for the function name used by this specific `CallExpression`. That probably means that you'll have to extend `getInlineableModule` to return the `requireFunctionName` to have it available here.